exceptions/exceptions.hh

```diff
@@ -228,6 +228,7 @@ struct read_failure_exception : public request_failure_exception {
 };
 
 struct overloaded_exception : public cassandra_exception {
+    overloaded_exception() noexcept : cassandra_exception(exception_code::OVERLOADED, "") {};
     explicit overloaded_exception(size_t c) noexcept :
```

Comment:
Not related to this patch, but since I'm looking at this code now, I want to comment that if we start using overloaded_exception more, we should get rid of this hints-specific constructor, and move this message to the hints code which uses it.